### PR TITLE
fix(fhir): quantity 0 results in no value in observations

### PR DESCRIPTION
**Pull request by jonahkaye**
Refs: #[2147](https://github.com/metriport/metriport-internal/issues/2147)

### Description

- quantities of 0 don't get mapped and then observations have no values in them

### Testing

- Local
  - [x] e2e test
  - [x] insert test? 

### Release Plan

- [ ] Merge this

### CODE REVIEW

**Review by leite08 — COMMENTED** [review]
Wouldn't it be easier to check if the return is empty instead of returning an object?

File: packages/fhir-converter/src/templates/cda/Resources/MedicationStatement.hbs [hunk 1]
```diff
@@ -45,13 +45,17 @@
...
-                        {{/with}}
+                        {{#if medicationStatement.doseQuantity.low}}
+                            "doseRange": {{>DataType/Range.hbs range=medicationStatement.doseQuantity}},
+                        {{else}}
+                            {{#with (extractRangeFromQuantity medicationStatement.doseQuantity) as |rangeObj|}}
+                                {{#if rangeObj.isValid}}
+                                    "doseRange": {{>DataType/Range.hbs range=rangeObj.range}},
+                                {{else}}
+                                    "doseQuantity": {{>DataType/Quantity.hbs quantity=../medicationStatement.doseQuantity}},
+                                {{/if}}
```
Comment (jonahkaye) [inline packages/fhir-converter/src/templates/cda/Resources/MedicationStatement.hbs:56]: Screenshot 2024-09-16 at 22 00 21

missing all doses that look like this :(

File: packages/fhir-converter/src/lib/handlebars-converter/handlebars-helpers.js [hunk 2]
```diff
@@ -1363,6 +1363,11 @@ module.exports.external = [
             const ret = cleanUpReturn(low, high, "low", "high");
             if (ret) return ret;
           }
+        } else if (range.low || range.high) {
+          const low = getRangeLimit(range.low);
+          const high = getRangeLimit(range.high);
+          const ret = cleanUpReturn(low, high, "low", "high");
```
Comment (jonahkaye) [inline packages/fhir-converter/src/lib/handlebars-converter/handlebars-helpers.js:1369]: this did not work for one of the most common dosage range structures :(

File: packages/fhir-converter/src/templates/cda/Resources/MedicationStatement.hbs (hunk 1 above)
Comment (leite08) [reply packages/fhir-converter/src/templates/cda/Resources/MedicationStatement.hbs:56]: 💪

**Review by leite08 — APPROVED** [review]

**Review by RamilGaripov — APPROVED** [review]

File: packages/fhir-converter/src/lib/handlebars-converter/handlebars-helpers.js (hunk 2 above)
Comment (RamilGaripov) [reply packages/fhir-converter/src/lib/handlebars-converter/handlebars-helpers.js:1369]: Can we figure this out so this patch goes out?

**Pull request merged**